README.md

```diff
@@ -1,34 +1,72 @@
+
+### Building images
+
+ubuntu-image offers two basic sub-commands for building snap-based and classical images.
+
+To build a snap-based image with ubuntu-image, you need a [model assertion](https://ubuntu.com/core/docs/reference/assertions/model). A model assertion is a YAML file that describes a particular combination of core, kernel, and gadget snaps, along with other declarations, signed with a digital signature asserting its authenticity. The `ubuntu-image` command only requires the path to this model assertion to build snap-based images.
+
+To build snap-based images with `ubuntu-image`, use the following command:
+
+```
+ubuntu-image snap model.assertion
+```
+
+See [Build your first Ubuntu Core image](https://ubuntu.com/core/docs/build-an-image) for more information on building snap-based images using ubuntu-image. To build an image with custom snaps, see [Build an image with custom snaps](https://ubuntu.com/core/docs/custom-images).
+
+Classical images are built from a local [gadget tree](https://github.com/snapcore/pc-gadget) path. The [gadget tree](https://github.com/snapcore/pc-gadget) is a primed [gadget snap](https://snapcraft.io/docs/the-gadget-snap) containing a `gadget.yaml` file in the meta directory and all the necessary bootloader gadget bits built. You can prepare a [gadget tree](https://github.com/snapcore/pc-gadget) by fetching a specially customised [gadget snap](https://snapcraft.io/docs/the-gadget-snap) source and running `snapcraft prime` on it. The resulting tree will be stored in the prime/ directory.
+
+The `gadget.yaml` file contains important information about the volumes that will be created. You can think of these volumes as disk images. The file also specifies the structures that will be included within each volume. Structures define the layout of the volume, including partitions, Master Boot Records, and any other relevant content. Also, the `gadget.yaml` file indicates whether a bootloader will be included in the volume and, if so, the type of bootloader it will be.
+
+To build classical images with ubuntu-image, use the following command:
+
+```
+ubuntu-image classic image_definition.yaml
+```
+
+For more information on how to build a gadget snap, see [Building a gadget snap](https://ubuntu.com/core/docs/gadget-building).
+
+## Building and testing ubuntu-image
+
+See [Contributing to ubuntu-image](https://github.com/canonical/ubuntu-image/blob/main/CONTRIBUTING.md) for instructions on how to set up, build, and test ubuntu-image in development mode.
+
+## License
+
+The ubuntu-image project is licensed under [GNU General Public License v3.0](https://github.com/canonical/ubuntu-image/blob/main/LICENSE).
+
+## Contributing to ubuntu-image
+
+To learn how to contribute to the ubuntu-image project, see [Contributing to ubuntu-image](https://github.com/canonical/ubuntu-image/blob/main/CONTRIBUTING.md).
```

Comment:
Any particular reason to use full URLs instead of relative paths?

Reply:
@upils no particular reason that I can think of. I can use relative paths if that won't adversely affect the links on [pkg.go.dev/github.com/canonical/ubuntu-image](https://pkg.go.dev/github.com/canonical/ubuntu-image).

Reply:
I expected relative links wouldn't work on pkg.go.dev.